framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1018,14 +1018,22 @@ public TransferResult<V, S> visitMethodInvocation(
         // add new information based on postcondition
         processPostconditions(n, store, method, invocationTree);
 
-        S thenStore = store;
```

Comment:
Looking through the test failures, I think the following might work better: check whether the return type of the method is `boolean` or `Boolean` (using [`TypesUtils.isBooleanType](https://github.com/eisop/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java#L335)).
If it is, do what the previous code did. Otherwise, return the RegularTransferResult.
Only methods with a boolean return can have a conditional postcondition and additionally, statically executable boolean functions also need multiple stores. To account for possible future extensions, let's just do this for all boolean functions.